### PR TITLE
add routes-admin branch for admin

### PR DESCRIPTION
**Pull request by HammamAbuShehadeh**
#9 add admin route

### CODE REVIEW

**Comment by HammamAbuShehadeh** [comment]
Done Update scheme models files

File: server/server/routes/adminRoute.js
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:1]: I will start review

File: server/server/routes/adminRoute.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const express = require("express");
+const router = express.Router();
+const { route } = require("..");
+const { ensureAuth, verifyIsAdmin } = require("../middlewares/userAuth");
+
+const adminController = require("../controllers/adminController");
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:6]: looks good import

File: server/server/routes/adminRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+// Define admin routes
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:8]: remove unnecessary comment

File: server/server/routes/adminRoute.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+// for get all dishes
+router.get("/Dishes", ensureAuth, verifyIsAdmin, adminController.getDishItems);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:11]: looks good

File: server/server/routes/adminRoute.js [hunk 4]
```diff
@@ -9,6 +9,13 @@ const adminController = require("../controllers/adminController");
 
 // for get all dishes
 router.get("/Dishes", ensureAuth, verifyIsAdmin, adminController.getDishItems);
+// for add new Restaurant
+router.post(
+  "/Restaurant",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.addRestaurant
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:18]: well done

File: server/server/routes/adminRoute.js [hunk 5]
```diff
@@ -16,6 +16,13 @@ router.post(
   verifyIsAdmin,
   adminController.addRestaurant
 );
+// add dish for selected Restaurant
+router.post(
+  "/DishToRestaurant",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.addDishToRestaurant
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js]: looks good

File: server/server/routes/adminRoute.js [hunk 6]
```diff
@@ -30,6 +30,13 @@ router.get(
   verifyIsAdmin,
   adminController.getAllRestaurants
 );
+// get dishes for one Restaurant
+router.get(
+  "/restaurants/:restaurantId",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.getAllDishes
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js]: well done

File: server/server/routes/adminRoute.js [hunk 7]
```diff
@@ -37,6 +37,14 @@ router.get(
...
   adminController.getAllDishes
 );
+// delete Restaurant or Restaurants
+router.delete(
+  "/restaurantDelete",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.deleteRestaurant
+);
+
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js]: no changes needed

File: server/server/routes/adminRoute.js [hunk 8]
```diff
@@ -44,6 +44,13 @@ router.delete(
   verifyIsAdmin,
   adminController.deleteRestaurant
 );
+// delete dish for specific Restaurant
+router.delete(
+  "/dishDelete",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.removeOneOrManyItems
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js]: looks good

File: server/server/routes/adminRoute.js [hunk 9]
```diff
@@ -51,6 +51,13 @@ router.delete(
   verifyIsAdmin,
   adminController.removeOneOrManyItems
 );
+// for search dishes minPrice and maxPrice and category and name
+router.get(
+  "/searchDishes",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.searchDishes
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:60]: make the comment more clear

File: server/server/routes/adminRoute.js [hunk 10]
```diff
@@ -58,6 +58,13 @@ router.get(
   verifyIsAdmin,
   adminController.searchDishes
 );
+//edit restaurant
+router.put(
+  "/restaurant/edit",
+  ensureAuth,
+  verifyIsAdmin,
+  adminController.editRestaurant
+);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:67]: add auth plz

File: server/server/routes/adminRoute.js [hunk 11]
```diff
@@ -65,6 +65,9 @@ router.put(
   verifyIsAdmin,
   adminController.editRestaurant
 );
+// edit dish
+router.put("/dish/edit", ensureAuth, verifyIsAdmin, adminController.editDish);
+
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:70]: looks good

File: server/server/routes/adminRoute.js [hunk 12]
```diff
@@ -68,7 +68,10 @@ router.put(
 // edit dish
 router.put("/dish/edit", ensureAuth, verifyIsAdmin, adminController.editDish);
 
-
+// Get past completed orders for report on admin
+router.get('/orders/past',
+//  isAuthenticated,
+   adminController.allPastOrders);
```
Comment (MonaAlHajEid) [inline server/server/routes/adminRoute.js:74]: add auth plz

**Review by MonaAlHajEid — COMMENTED** [review]
good job, plz review the comments ^^

File: server/server/routes/adminRoute.js
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:1]: sure

File: server/server/routes/adminRoute.js (hunk 1 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:6]: thx

File: server/server/routes/adminRoute.js (hunk 2 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:8]: sure i will do it later

File: server/server/routes/adminRoute.js (hunk 4 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:18]: thanks a lot

File: server/server/routes/adminRoute.js (hunk 6 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js]: thx

File: server/server/routes/adminRoute.js (hunk 9 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:60]: sure i do it later

File: server/server/routes/adminRoute.js (hunk 10 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:67]: sure i do it later

File: server/server/routes/adminRoute.js (hunk 12 above)
Comment (HammamAbuShehadeh) [reply server/server/routes/adminRoute.js:74]: sure i will do it

**Pull request merged**